packages/flutter/lib/src/rendering/sliver_fill.dart

```diff
@@ -83,21 +83,36 @@ class RenderSliverFillRemainingWithScrollable extends RenderSliverSingleBoxAdapt
     final SliverConstraints constraints = this.constraints;
     final double extent = constraints.remainingPaintExtent - math.min(constraints.overlap, 0.0);
 
+    final double cacheExtent = calculateCacheOffset(
+      constraints,
+      from: 0.0,
+      to: constraints.viewportMainAxisExtent,
+    );
     if (child != null) {
+      double maxExtent = extent;
+
+      // If sliver has no extent, but is within viewport's cacheExtent, use the
+      // sliver's cacheExtent as the maxExtent so that it does not get dropped
+      // from the semantic tree.
+      if (extent == 0 && cacheExtent > 0) {
+        maxExtent = cacheExtent;
+      }
       child!.layout(constraints.asBoxConstraints(
         minExtent: extent,
-        maxExtent: extent,
+        maxExtent: maxExtent,
       ));
     }
 
     final double paintedChildSize = calculatePaintOffset(constraints, from: 0.0, to: extent);
     assert(paintedChildSize.isFinite);
     assert(paintedChildSize >= 0.0);
+
     geometry = SliverGeometry(
       scrollExtent: constraints.viewportMainAxisExtent,
       paintExtent: paintedChildSize,
       maxPaintExtent: paintedChildSize,
       hasVisualOverflow: extent > constraints.remainingPaintExtent || constraints.scrollOffset > 0.0,
+      cacheExtent: cacheExtent,
```

Comment:
Should `cacheExtent` be updated to reflect what extent the child ended up being after laying out?

Reply:
See comment above: https://github.com/flutter/flutter/pull/143612/files#r1502876031